mesonbuild/modules/gnome.py

```diff
@@ -884,12 +884,16 @@ def gdbus_codegen(self, state, args, kwargs):
         xml_files = args[1:]
         target_name = namebase + '-gdbus'
         cmd = [self.interpreter.find_program_impl('gdbus-codegen')]
+        extra_args = mesonlib.stringlistify(kwargs.pop('extra_args', []))
+        cmd += extra_args
         if 'interface_prefix' in kwargs:
             cmd += ['--interface-prefix', kwargs.pop('interface_prefix')]
         if 'namespace' in kwargs:
             cmd += ['--c-namespace', kwargs.pop('namespace')]
         if kwargs.get('object_manager', False):
             cmd += ['--c-generate-object-manager']
+        if 'autocleanup' in kwargs:
+            cmd += ['--c-generate-autocleanup', kwargs.pop('autocleanup')]
```

Comment:
If the arg can have only one of two values, the code should check for that here and error out with a readable error message if the value is incorrect.